From 0f502ff64870abc16ef0eb8ec4a4e146a05d4528 Mon Sep 17 00:00:00 2001
From: zsy-jason <186247581+zsy-jason@users.noreply.github.com>
Date: Mon, 3 Nov 2025 21:52:11 +0800
Subject: [PATCH] [Infra] Support system expat.

---
 BUILD.gn | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/BUILD.gn b/BUILD.gn
index 78ddb05..eac1ab6 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -2,7 +2,11 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-if (is_linux) {
+declare_args() {
+  use_system_expat = is_linux
+}
+
+if (use_system_expat) {
   config("expat_config") {
     libs = [ "expat" ]
   }
-- 
2.49.0

