From d185332c7acc5986083e08799716fb7fa0863ec8 Mon Sep 17 00:00:00 2001
From: zsy-jason <186247581+zsy-jason@users.noreply.github.com>
Date: Wed, 11 Mar 2026 14:33:49 +0800
Subject: [PATCH] optimize: rename modp method with lynx prefix to avoid symbol
 conflict.

---
 modp_b64.cc |  8 ++++----
 modp_b64.h  | 52 ++++++++++++++++++++++++++--------------------------
 2 files changed, 30 insertions(+), 30 deletions(-)

diff --git a/modp_b64.cc b/modp_b64.cc
index 9be5820..b72d351 100644
--- a/modp_b64.cc
+++ b/modp_b64.cc
@@ -49,7 +49,7 @@
 
 #define BADCHAR 0x01FFFFFF
 
-size_t modp_b64_encode_data(char* dest, const char* str, size_t len)
+size_t lynx_modp_b64_encode_data(char* dest, const char* str, size_t len)
 {
     size_t i = 0;
     uint8_t* p = (uint8_t*) dest;
@@ -88,8 +88,8 @@ size_t modp_b64_encode_data(char* dest, const char* str, size_t len)
     return p - (uint8_t*)dest;
 }
 
-size_t modp_b64_encode(char* dest, const char* str, size_t len) {
-  size_t output_size = modp_b64_encode_data(dest, str, len);
+size_t lynx_modp_b64_encode(char* dest, const char* str, size_t len) {
+  size_t output_size = lynx_modp_b64_encode_data(dest, str, len);
   dest[output_size] = '\0';
   return output_size;
 }
@@ -115,7 +115,7 @@ size_t do_decode_padding(const char* src, size_t len, ModpDecodePolicy policy) {
   return len % 4 == 1 ? MODP_B64_ERROR : len;
 }
 
-size_t modp_b64_decode(char* dest, const char* src, size_t len, ModpDecodePolicy policy)
+size_t lynx_modp_b64_decode(char* dest, const char* src, size_t len, ModpDecodePolicy policy)
 {
     if (len != 0)
       len = do_decode_padding(src, len, policy);
diff --git a/modp_b64.h b/modp_b64.h
index 0d0612e..771a108 100644
--- a/modp_b64.h
+++ b/modp_b64.h
@@ -36,7 +36,7 @@ extern "C" {
  * src contains the bytes
  * len contains the number of bytes in the src
  * dest should be allocated by the caller to contain
- *   at least modp_b64_encode_len(len) bytes (see below)
+ *   at least lynx_modp_b64_encode_len(len) bytes (see below)
  *   This will contain the (non-null terminated) b64 bytes.
  * returns length of the destination string.
  *
@@ -45,8 +45,8 @@ extern "C" {
  * \code
  * char* src = ...;
  * int srclen = ...; //the length of number of bytes in src
- * char* dest = (char*) malloc(modp_b64_encode_len);
- * int len = modp_b64_encode_data(dest, src, sourcelen);
+ * char* dest = (char*) malloc(lynx_modp_b64_encode_len);
+ * int len = lynx_modp_b64_encode_data(dest, src, sourcelen);
  * if (len == -1) {
  *   printf("Error\n");
  * } else {
@@ -55,18 +55,18 @@ extern "C" {
  * \endcode
  *
  */
-size_t modp_b64_encode_data(char* dest, const char* str, size_t len);
+size_t lynx_modp_b64_encode_data(char* dest, const char* str, size_t len);
 
 /**
- * Same as modp_b64_encode_data, but additionally sets a null terminator at the
+ * Same as lynx_modp_b64_encode_data, but additionally sets a null terminator at the
  * end of `dest` (i.e. at dest[output_size]).
- * Like modp_b64_encode_data, returns the length of the destination string (i.e.
+ * Like lynx_modp_b64_encode_data, returns the length of the destination string (i.e.
  * not counting the null terminator).
  *
  * TODO(csharrison): Consider removing this once all callers migrate to
- * modp_b64_encode_data.
+ * lynx_modp_b64_encode_data.
  */
-size_t modp_b64_encode(char* dest, const char* str, size_t len);
+size_t lynx_modp_b64_encode(char* dest, const char* str, size_t len);
 
 /**
  * Decode a base64 encoded string
@@ -84,8 +84,8 @@ size_t modp_b64_encode(char* dest, const char* str, size_t len);
  * \code
  * char* src = ...;
  * int srclen = ...; // or if you don't know use strlen(src)
- * char* dest = (char*) malloc(modp_b64_decode_len(srclen));
- * int len = modp_b64_decode(dest, src, sourcelen);
+ * char* dest = (char*) malloc(lynx_modp_b64_decode_len(srclen));
+ * int len = lynx_modp_b64_decode(dest, src, sourcelen);
  * if (len == -1) { error }
  * \endcode
  */
@@ -101,32 +101,32 @@ enum class ModpDecodePolicy {
   // Accepts any number of pad chars.
   kNoPaddingValidation,
 };
-size_t modp_b64_decode(
+size_t lynx_modp_b64_decode(
     char* dest,
     const char* src,
     size_t len,
     ModpDecodePolicy policy = ModpDecodePolicy::kStrict);
 
 /**
- * The maximum input that can be passed into modp_b64_encode{_data}.
- * Lengths beyond this will overflow modp_b64_encode_len.
+ * The maximum input that can be passed into lynx_modp_b64_encode{_data}.
+ * Lengths beyond this will overflow lynx_modp_b64_encode_len.
  *
- * This works because modp_b64_encode_len(A) computes:
+ * This works because lynx_modp_b64_encode_len(A) computes:
  *     ceiling[max_len / 3] * 4 + 1
  *   = ceiling[floor[(SIZE_MAX-1)/4]*3 / 3] * 4 + 1
  *   = floor[(SIZE_MAX-1)/4] * 4 + 1
  *  <= SIZE_MAX-1 + 1
  *   = SIZE_MAX
  *
- * Note: technically modp_b64_encode_data can take one extra byte, but for
+ * Note: technically lynx_modp_b64_encode_data can take one extra byte, but for
  * simplicity the bound is shared between the two functions.
  */
 #define MODP_B64_MAX_INPUT_LEN ((SIZE_MAX - 1) / 4 * 3)
 
 /**
  * Given a source string of length len, this returns the amount of
- * memory the destination string should have, for modp_b64_encode_data and
- * modp_b64_encode, respectively.
+ * memory the destination string should have, for lynx_modp_b64_encode_data and
+ * lynx_modp_b64_encode, respectively.
  *
  * remember, this is integer math
  * 3 bytes turn into 4 chars
@@ -136,8 +136,8 @@ size_t modp_b64_decode(
  * WARNING: These expressions will overflow if the A is above
  * MODP_B64_MAX_INPUT_LEN. The caller must check this bound first.
  */
-#define modp_b64_encode_data_len(A) ((A + 2) / 3 * 4)
-#define modp_b64_encode_len(A) (modp_b64_encode_data_len(A) + 1)
+#define lynx_modp_b64_encode_data_len(A) ((A + 2) / 3 * 4)
+#define lynx_modp_b64_encode_len(A) (lynx_modp_b64_encode_data_len(A) + 1)
 
 /**
  * Given a base64 string of length len,
@@ -148,7 +148,7 @@ size_t modp_b64_decode(
  * decode  4 chars turn into 3 bytes
  * floor[len * 3/4] + 2
  */
-#define modp_b64_decode_len(A) (A / 4 * 3 + 2)
+#define lynx_modp_b64_decode_len(A) (A / 4 * 3 + 2)
 
 #define MODP_B64_ERROR ((size_t)-1)
 
@@ -156,10 +156,10 @@ size_t modp_b64_decode(
 }
 
 #include <string>
-inline std::string& modp_b64_encode(std::string& s)
+inline std::string& lynx_modp_b64_encode(std::string& s)
 {
-    std::string x(modp_b64_encode_len(s.size()), '\0');
-    size_t d = modp_b64_encode(const_cast<char*>(x.data()), s.data(), (int)s.size());
+    std::string x(lynx_modp_b64_encode_len(s.size()), '\0');
+    size_t d = lynx_modp_b64_encode(const_cast<char*>(x.data()), s.data(), (int)s.size());
     x.erase(d, std::string::npos);
     s.swap(x);
     return s;
@@ -173,10 +173,10 @@ inline std::string& modp_b64_encode(std::string& s)
  * \param[in,out] s the string to be decoded
  * \return a reference to the input string
  */
-inline std::string& modp_b64_decode(std::string& s)
+inline std::string& lynx_modp_b64_decode(std::string& s)
 {
-    std::string x(modp_b64_decode_len(s.size()), '\0');
-    size_t d = modp_b64_decode(const_cast<char*>(x.data()), s.data(), (int)s.size());
+    std::string x(lynx_modp_b64_decode_len(s.size()), '\0');
+    size_t d = lynx_modp_b64_decode(const_cast<char*>(x.data()), s.data(), (int)s.size());
     if (d == MODP_B64_ERROR) {
         x.clear();
     } else {
-- 
2.49.0

